src/components/Images/Images.js

```diff
@@ -1,64 +1,347 @@
+    ...baseImageSettings,
+    ...(!!width[2]) && { width: width[2]},
+    ...(!!height[2]) && { height: height[2]},
+  }
+
+  const laptopUrl = cld.url(publicIdFilename, laptopImageSettings)
+
+  const laptopImageWebP = {
+    ...laptopImageSettings,
+    format: 'webp'
+  }
+
+  const laptopImageJp2 = {
+    ...laptopImageSettings,
+    format: 'jp2'
+  }
+
+  const laptopImageJpeg = {
+    ...laptopImageSettings,
+    format: 'jpeg'
+  }
+
+  const laptopUrlWebP = cld.url(publicIdFilename, {...laptopImageWebP, ...dpr1Setting})
+  const laptopUrlJp2 = cld.url(publicIdFilename, {...laptopImageJp2, ...dpr1Setting})
+  const laptopUrlJpeg = cld.url(publicIdFilename, {...laptopImageJpeg, ...dpr1Setting})
+
+  const laptopUrlWebP_2x = cld.url(publicIdFilename, {...laptopImageWebP, ...dpr2Setting})
+  const laptopUrlJp2_2x = cld.url(publicIdFilename, {...laptopImageJp2, ...dpr2Setting})
+  const laptopUrlJpeg_2x = cld.url(publicIdFilename, {...laptopImageJpeg, ...dpr2Setting})
+
+  const laptopUrlWebP_3x = cld.url(publicIdFilename, {...laptopImageWebP, ...dpr3Setting})
+  const laptopUrlJp2_3x = cld.url(publicIdFilename, {...laptopImageJp2, ...dpr3Setting})
+  const laptopUrlJpeg_3x = cld.url(publicIdFilename, {...laptopImageJpeg, ...dpr3Setting})
+
+  // ------------------------------
+  // Desktop Settings
+  // ------------------------------
+  const desktopImageSettings = {
+    ...baseImageSettings,
+    ...(!!width[3]) && { width: width[3]},
+    ...(!!height[3]) && { height: height[3]},
+  }
+
+  const desktopUrl = cld.url(publicIdFilename, desktopImageSettings)
+
+  const desktopImageWebP = {
+    ...desktopImageSettings,
+    format: 'webp'
+  }
+
+  const desktopImageJp2 = {
+    ...desktopImageSettings,
+    format: 'jp2'
+  }
+
+  const desktopImageJpeg = {
+    ...desktopImageSettings,
+    format: 'jpeg'
+  }
+
+  const desktopUrlWebP = cld.url(publicIdFilename, {...desktopImageWebP, ...dpr1Setting})
+  const desktopUrlJp2 = cld.url(publicIdFilename, {...desktopImageJp2, ...dpr1Setting})
+  const desktopUrlJpeg = cld.url(publicIdFilename, {...desktopImageJpeg, ...dpr1Setting})
+
+  const desktopUrlWebP_2x = cld.url(publicIdFilename, {...desktopImageWebP, ...dpr2Setting})
+  const desktopUrlJp2_2x = cld.url(publicIdFilename, {...desktopImageJp2, ...dpr2Setting})
+  const desktopUrlJpeg_2x = cld.url(publicIdFilename, {...desktopImageJpeg, ...dpr2Setting})
+
+  const desktopUrlWebP_3x = cld.url(publicIdFilename, {...desktopImageWebP, ...dpr3Setting})
+  const desktopUrlJp2_3x = cld.url(publicIdFilename, {...desktopImageJp2, ...dpr3Setting})
+  const desktopUrlJpeg_3x = cld.url(publicIdFilename, {...desktopImageJpeg, ...dpr3Setting})
+
   return (
-    <Cloudinary.Image
-      {...rest}
-      className={['Image', className].join(' ')}
-      cloudName={CLOUDINARY_CLOUD_NAME}
-      dpr="auto"
-      responsive
-      publicId={publicIdFilename}
-      crop={crop} 
-      secure="true"
-      flags="progressive:semi"
-    />
+    <picture>
```

Comment:
Wow, they make us configure all this? Seems like they'd let us setup some sort of simpler metadata object and just initialize and be done 🤔

Reply:
Each call to their Image Transformation API relies on building a dynamic URL with parameters specific to the resource you're requesting. 

There wasn't an option I could see in the SDK for generating a `<picture>`  tag with `<source>` tags automatically.

Reply:
Are you referring to setting up all the different `<source>` tags manually @roblevintennis or something to do with the options we pass when generating each URL?